buildroot-external/board/hardkernel/odroid-m1s/patches/uboot/0004-Fix-ethernet-properties.patch

```diff
@@ -0,0 +1,52 @@
+From 65a304151817626157e7be4e7e1581cdddca2a02 Mon Sep 17 00:00:00 2001
+From: Tim Lunn <tim@feathertop.org>
+Date: Sat, 3 Feb 2024 23:42:47 +1100
+Subject: [PATCH] Fix ethernet properties
```

Comment:
It is needed to match up with `linux/0005-Remove-deprecated-snps-reset-properties.patch`

The snps,reset properties under &gmac are deprecated and moved to properties under &mdio (see above patch). However with this the kernel is unable to reset the ethernet phy correctly and requires the reset to happen within uboot. There was a long discussion upstream on linux list about chicken+egg problems with the reset, where kernel can't reset until it get id, but can't get id until reset or something. So Kwiboo fixed it via uboot resetting.

This is inline with how the Odroid-m1 is handled upstream and probably all other rk356x boards now. Green would also need these same changes at some point, prior to upstreaming.

Reply:
Is that maybe also related to what I've seen on my ODROID-M1? I've had to disable Ethernet in U-Boot in this case though, see #3122.

Reply:
I suppose it could be related, odroid M1 is already using the new properties though (atleast in 6.6 / 2024.01). However I can't actually find the upstream thread discussing this now.